fly/proxy.ts

```diff
@@ -112,7 +171,28 @@ export class FlyProxy {
     const region =
       regionsToTry[Math.floor(Math.random() * regionsToTry.length)];
 
-    const createMachinePayload = {
+    let volumeId;
+    if (storageInGB) {
+      try {
+        volumeId = await this.createVolume(region, storageInGB);
+      } catch (e) {
+        console.error("Failed to create volume", e);
+        return this.startMachineInner(
+          namePrefix,
+          image,
+          cpus,
+          memory,
+          storageInGB,
+          env,
+          attempts + 1,
+          // remove the region we just tried from the list
+          regionsToTry.filter((r) => r !== region),
```

Comment:
Why would this fail in certain regions?

Reply:
Sometimes fly has capacity issues. This actually hasn't happened in a while but over the summer it was pretty frequent.

Reply:
Ah got it